### PR TITLE
Added options for harvested and selected records and output of xml records

### PR DESCRIPTION
**Pull request by sachitrajbhandari**
https://github.com/aodn/backlog/issues/1643

### CODE REVIEW

File: mafia/src/main/java/au/org/emii/PostgresEditor.java [hunk 1]
```diff
@@ -68,6 +58,12 @@
 class MyValidationErrorHandler implements ErrorHandler {
 
     // pass the uuid in to output better errors...
```
Comment (jonescc) [inline mafia/src/main/java/au/org/emii/PostgresEditor.java]: This comment no longer seems to relate to the code.

Should be a TODO?

File: mafia/src/main/java/au/org/emii/PostgresEditor.java [hunk 2]
```diff
@@ -218,6 +249,11 @@ public static void main(String[] args) throws Exception {
             transformer = identity;
         }
 
+        // Empty invalid records list file
+        if(cmd.hasOption("invalids")) {
+            new FileWriter(cmd.getOptionValue("invalids")).close();
+        }
+
         // TODO this crap really wants to be factored, so we pass the action and the transformer in...
```
Comment (jonescc) [inline mafia/src/main/java/au/org/emii/PostgresEditor.java:257]: Gee Julian was really critical of his own work!

File: README.md [hunk 3]
```diff
@@ -33,19 +33,24 @@ concurrent updates by the application
...
+ -help                  show help
+ -pass <arg>            password
+ -stdout                dump result to stdout
+ -transform <arg>       transform stylesheet (.xslt) file to use
+ -update                perform inplace update of the metadata record
+ -url <arg>             jdbc connection string, eg.
+                        jdbc:postgresql://127.0.0.1/geonetwork
+ -user <arg>            user
+ -uuid <arg>            etl applies to specific metadata record
+- selected <arg>        etl applies to selected metadata record uuids listed in the provided file
```
Comment (jonescc) [inline README.md]: need to remove space between '-' and options?

File: mafia/src/main/java/au/org/emii/PostgresEditor.java [hunk 4]
```diff
@@ -158,6 +177,8 @@ public static void main(String[] args) throws Exception {
         // selection
         options.addOption("uuid", true, "etl applies to specific metadata record");
         options.addOption("all", false, "etl applies to all metadata records");
+        options.addOption("selected", true, "etl applies to selected metadata records");
+        options.addOption("isharvested", true, "etl applies to harvested or non-harvested metadata records depending upon provided value (y or n)");
```
Comment (jonescc) [inline mafia/src/main/java/au/org/emii/PostgresEditor.java:181]: whats the default?

**Review by jonescc — COMMENTED** [review]
Looks good - some tidy up comments only

File: mafia/src/main/java/au/org/emii/PostgresEditor.java (hunk 4 above)
Comment (jonescc) [reply mafia/src/main/java/au/org/emii/PostgresEditor.java:181]: I'd make the default non-harvested records only.  

We don't ever really want to update harvested records except when we are using them for  testing purposes.

Comment (sachitrajbhandari) [reply mafia/src/main/java/au/org/emii/PostgresEditor.java:181]: with option isharvested
- isharvested=y : harvested records only
- isharversted=n : non-harversted records only

without option isharvested:
- all records (both harvested and non-harvested)

Comment (jonescc) [reply mafia/src/main/java/au/org/emii/PostgresEditor.java:181]: Will do for the moment

**Review by jonescc — APPROVED** [review]

File: mafia/src/main/java/au/org/emii/PostgresEditor.java (hunk 1 above)
Comment (jonescc) [reply mafia/src/main/java/au/org/emii/PostgresEditor.java]: Changed to TODO

File: README.md (hunk 3 above)
Comment (jonescc) [reply README.md]: Fixed

**Pull request merged**